huggingface_pipelines/text.py

```diff
@@ -0,0 +1,599 @@
+
+    Example:
+        config = TextSegmentationPipelineConfig(
+            columns=['text'], output_path='./output')
+        pipeline = TextSegmentationPipeline(config)
+        result = pipeline(
+            {'text': ['This is a sample text.', 'Another example.']})
+    """
+
+    SPACY_MODELS = {
+        "eng_Latn": "en_core_web_sm",
+        "fra_Latn": "fr_core_news_sm",
+        "deu_Latn": "de_core_news_sm",
+        "spa_Latn": "es_core_news_sm",
+        "ita_Latn": "it_core_news_sm",
+        "por_Latn": "pt_core_news_sm",
+        "nld_Latn": "nl_core_news_sm",
+    }
+
+    config: TextSegmentationPipelineConfig
+
+    def __init__(self, config: TextSegmentationPipelineConfig):
+        """
+        Initialize the TextSegmentationPipeline with the given configuration.
+
+        Args:
+            config (TextSegmentationPipelineConfig): Configuration for the text segmentation pipeline.
+        """
+        super().__init__(config)
+        self.nlp = self.load_spacy_model(self.config.source_lang)
+        logger.info("Text preprocessing model initialized.")
+
+    def load_spacy_model(self, lang_code: str) -> Language:
+        """
+        Loads the appropriate spaCy model based on the language code.
+
+        Args:
+            lang_code (str): The language code for the desired spaCy model.
+
+        Returns:
+            Language: The loaded spaCy language model.
+
+        Raises:
+            ValueError: If the language code is not supported.
+
+        Example:
+            nlp = pipeline.load_spacy_model('en')
+        """
+        if lang_code not in self.SPACY_MODELS:
+            raise ValueError(f"No installed model found for language code: {lang_code}")
+        return spacy.load(self.SPACY_MODELS[lang_code])
+
+    def segment_text(self, text: Optional[str]) -> List[str]:
+        """
+        Preprocesses a single text by segmenting it into sentences.
+
+        Args:
+            text (Optional[str]): The input text to preprocess.
+
+        Returns:
+            List[str]: A list of preprocessed sentences.
+
+        Raises:
+            ValueError: If an invalid handle_missing option is specified in the configuration.
+
+        Example:
+            sentences = pipeline.segment_text(
+                "This is a sample. It has two sentences.")
+            print(sentences)  # ['This is a sample.', 'It has two sentences.']
+        """
+        if text is None or (isinstance(text, str) and text.strip() == ""):
+            if self.config.handle_missing == "skip":
+                return []
+            elif self.config.handle_missing == "remove":
+                return []
+            elif self.config.handle_missing == "fill":
+                return [self.config.fill_value] if self.config.fill_value else []
+            else:
+                raise ValueError(
+                    f"Invalid handle_missing option: {self.config.handle_missing}"
+                )
+
+        doc = self.nlp(text)
```

Comment:
By default, Spacy does a lot of expensive text analysis: POS tagging, syntactic parsing, named entity recognition etc. 
To speed up the pipeline, you may want to turn them off. Here https://github.com/explosion/spaCy/issues/453 is a discussion of how it might be done, but I am not sure if it is up to date.

Reply:
P.S. Maybe, you want to keep syntactic parsing, as the sentencization algorithm may depend on it. Need to dig deeper.